core/src/main/scala/org/http4s/Message.scala

```diff
@@ -189,6 +189,24 @@ case class Request(
   }
 
   def serverSoftware: ServerSoftware = attributes.get(Keys.ServerSoftware).getOrElse(ServerSoftware.Unknown)
+
+  /** Helper method for decoding [[Request]]s
+    *
+    * Attempt to decode the [[Request]] and, if successful, execute the continuation to get a [[Response]].
+    * If decoding fails, a BadRequest [[Response]] is generated.
+    */
+  def decode[T](f: T => Task[Response])(implicit decoder: EntityDecoder[T]): Task[Response] =
+    decoder.decode(this).fold(
+      e => Response(Status.BadRequest, httpVersion).withBody(e.sanitized),
+      f
+    ).join
+
+  /** Like [[decode]], but with an explicit decoder. */
+  def decodeWith[T](decoder: EntityDecoder[T])(f: T => Task[Response]): Task[Response] =
```

Comment:
It looks like one can be implemented in terms of the other, they are just aliases.